pkg/kubelet/certificate/bootstrap/bootstrap.go

```diff
@@ -207,3 +216,51 @@ func verifyKeyData(data []byte) bool {
 	_, err := certutil.ParsePrivateKeyPEM(data)
 	return err == nil
 }
+
+func waitForServer(rawurl string, retries int) error {
+	u, err := url.Parse(rawurl)
+	if err != nil {
+		return err
+	}
+	if u.Scheme != "http" && u.Scheme != "https" {
+		return fmt.Errorf("scheme must be 'http' or 'https' but got: %q", u.Scheme)
+	}
+
+	var d net.Dialer
```

Comment:
I'd just use `net.Dialer.Timeout` instead of contexts.
Also, does apiserver have some kind of `/health` endpoint? Maybe we should use that instead of plain reachability.